src/sd.jl

```diff
@@ -415,3 +415,141 @@ function build_constraint(
         shape,
     )
 end
+
+"""
+    HermitianPSDCone
+
+Hermitian positive semidefinite cone object that can be used to create a
+hermitian positive semidefinite square matrix in the [`@variable`](@ref)
+macro.
+
+## Examples
+
+Consider the following example:
+```jldoctest PSDCone; setup = :(using JuMP)
+julia> model = Model();
+
+julia> @variable(model, H[1:3, 1:3] in HermitianPSDCone())
+3×3 Matrix{GenericAffExpr{ComplexF64, VariableRef}}:
+ real(H[1,1])                                real(H[1,2]) + (0.0 + 1.0im) imag(H[1,2])   real(H[1,3]) + (0.0 + 1.0im) imag(H[1,3])
+ real(H[1,2]) + (-0.0 - 1.0im) imag(H[1,2])  real(H[2,2])                                real(H[2,3]) + (0.0 + 1.0im) imag(H[2,3])
+ real(H[1,3]) + (-0.0 - 1.0im) imag(H[1,3])  real(H[2,3]) + (-0.0 - 1.0im) imag(H[2,3])  real(H[3,3])
+```
+We see in the output of the last command that the matrix the vectorization of the
+matrix is constrained to belong to the `PositiveSemidefiniteConeSquare`.
```

Comment:
This sentence does't follow what I see. We should explain that `H` is an affine expression, but it is backed by a PSD matrix.